### ISSUE TITLE
Devhub V3: update gatsby lunr plugin search index to reference markdown nodes/events

### ISSUE THREAD

**Issue by patricksimonian** [bcgov/devhub-app-web#742]
The search index should have a single source of truth for documentation resources which will be a MarkdownRemark gatsby node. 

The great thing with this is that all MarkdownRemark nodes have a parent that they will reference. 

In addition we should index event nodes and have a way of surfacing them in parallel with markdown nodes during searches (these are the trials and tribulations of working with static content :) ) 

As a node any work should be branched from `imp/gatsby-source-github-raw-integration` and merged back into that branch! sorry for the ugly name. 

> why using an integration branch? Technically imp/gatsby-source-github-raw-integration can be merged into master as its features are not being used within the page components and it supports backwards compatibility with siphon, however because this is such a __large__ change, i felt throwing in code to production/master that may be stripped out later was not such a good deal
